### PR TITLE
OP-1325 OP-1370 Make attributes initialized via DI (Dependency Injection) "final"

### PR DESCRIPTION
**Pull request by mwithi**
See OP-1325.

Also:
- Made explicit `protected` modifier (it is default anyway)
- Removed one unused field
- Fixed formatting
- Fix failing tests (OP-1370)

### CODE REVIEW

File: src/main/java/org/isf/accounting/manager/BillBrowserManager.java [hunk 1]
```diff
@@ -108,52 +110,56 @@ public List<BillItems> getItems(int billID) throws OHServiceException {
...
 	 */
 	public List<BillPayments> getPayments(int billID) throws OHServiceException {
 		return ioOperations.getPayments(billID);
 	}
-	
+
 	/**
 	 * Stores a new {@link Bill} along with all its {@link BillItems} and {@link BillPayments}
+	 * 
 	 * @param bill - the bill to store.
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/accounting/manager/BillBrowserManager.java]: Technically the `-` isn't needed as the javadoc provides it; to if it is specified here then it is doubled.
```suggestion
	 * @param bill the bill to store.
```

**Review by dbmalkovsky — DISMISSED** [review]
LGTM given my 2 comments which can be ignored:

1. my comment about `-` in @param
2. my dislike for adding `final` when it really doesn't add anything

**Comment by mwithi** [comment]
> LGTM given my 2 comments which can be ignored:
> 
> 1. my comment about `-` in @param
> 2. my dislike for adding `final` when it really doesn't add anything

Ok, thanks. The reasons are explained in the issue itself OP-1325. Don't you agree with some of those?

**Comment by dbmalkovsky** [comment]
Yes they make sense.  I just don't find them all that important given the larger scope of things.

File: src/main/java/org/isf/accounting/manager/BillBrowserManager.java [hunk 2]
```diff
@@ -108,52 +110,56 @@ public List<BillItems> getItems(int billID) throws OHServiceException {
...
 		return ioOperations.getPayments(billID);
 	}
-	
+
 	/**
 	 * Stores a new {@link Bill} along with all its {@link BillItems} and {@link BillPayments}
-	 * @param bill - the bill to store.
+	 * 
+	 * @param bill the bill to store.
 	 * @param billItems - the list of bill's items
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/accounting/manager/BillBrowserManager.java]:
```suggestion
	 * @param billItems the list of bill's items
```

File: src/main/java/org/isf/accounting/manager/BillBrowserManager.java [hunk 3]
```diff
@@ -108,52 +110,56 @@ public List<BillItems> getItems(int billID) throws OHServiceException {
...
 	 * @param billPayments - the list of bill's payments
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/accounting/manager/BillBrowserManager.java]:
```suggestion
	 * @param billPayments the list of bill's payments
```

File: src/main/java/org/isf/accounting/manager/BillBrowserManager.java [hunk 4]
```diff
@@ -178,37 +185,40 @@ private Bill newBill(Bill newBill) throws OHServiceException {
...
 	 */
 	private void newBillPayments(int billID, List<BillPayments> payItems) throws OHServiceException {
 		ioOperations.newBillPayments(ioOperations.getBill(billID), payItems);
 	}
-	
+
 	/**
 	 * Updates the specified {@link Bill} along with all its {@link BillItems} and {@link BillPayments}
+	 * 
 	 * @param updateBill - the bill to update.
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/accounting/manager/BillBrowserManager.java]:
```suggestion
	 * @param updateBill the bill to update.
```

File: src/main/java/org/isf/accounting/manager/BillBrowserManager.java [hunk 5]
```diff
@@ -178,37 +185,40 @@ private Bill newBill(Bill newBill) throws OHServiceException {
...
 	 * @param billItems - the list of bill's items
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/accounting/manager/BillBrowserManager.java]:
```suggestion
	 * @param billItems the list of bill's items
```

File: src/main/java/org/isf/accounting/manager/BillBrowserManager.java [hunk 6]
```diff
@@ -178,37 +185,40 @@ private Bill newBill(Bill newBill) throws OHServiceException {
...
 	 * @param billPayments - the list of bill's payments
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/accounting/manager/BillBrowserManager.java]:
```suggestion
	 * @param billPayments the list of bill's payments
```
And there are about 60+ more like this

Comment (mwithi) [reply src/main/java/org/isf/accounting/manager/BillBrowserManager.java]: anyway this is the way I see the javadoc, doesn't look so bad...

image

Comment (mwithi) [reply src/main/java/org/isf/accounting/manager/BillBrowserManager.java]: anyway, I took the chance to revise some of them

**Comment by dbmalkovsky** [comment]
Using:   /home/dbm/.sdkman/candidates/java/17.0.10-zulu/bin/javadoc -protected -splitindex -d /home/dbm/dev/JavaDoc @/tmp/javadoc_args

Doesn't mean it works that way everywhere:
image

**Comment by mwithi** [comment]
> Using: /home/dbm/.sdkman/candidates/java/17.0.10-zulu/bin/javadoc -protected -splitindex -d /home/dbm/dev/JavaDoc @/tmp/javadoc_args
> 
> Doesn't mean it works that way everywhere

Agreed! Less is more!

**Comment by mwithi** [comment]
Note: for the failing cheks I've created OP-1370.

**Review by dbmalkovsky — APPROVED** [review]
LGTM.

**Pull request merged**